components/lib/push-firebase/src/main/java/mozilla/components/lib/push/firebase/AbstractFirebasePushService.kt

```diff
@@ -0,0 +1,61 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ *  License, v. 2.0. If a copy of the MPL was not distributed with this
+ *  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+package mozilla.components.lib.push.firebase
+
+import android.content.Context
+import com.google.android.gms.common.ConnectionResult
+import com.google.android.gms.common.GoogleApiAvailability
+import com.google.firebase.iid.FirebaseInstanceId
+import com.google.firebase.messaging.FirebaseMessaging
+import com.google.firebase.messaging.FirebaseMessagingService
+import com.google.firebase.messaging.RemoteMessage
+import mozilla.components.concept.push.EncryptedPushMessage
+import mozilla.components.concept.push.PushProvider
+import mozilla.components.concept.push.PushService
+
+abstract class AbstractFirebasePushService : FirebaseMessagingService(), PushService {
+
+    init {
+        start() // Allow the app to choose when to start? This is harder to allow than it looks..
+    }
+
+    final override fun start() {
+        FirebaseMessaging.getInstance().isAutoInitEnabled = true
```

Comment:
From the docs: `By default, FCM auto-initialization is enabled.` - unless you say otherwise in the manifest. So this seems like a no-op!